x-pack/plugins/ml/public/application/datavisualizer/data_drift/data_view_editor.tsx

```diff
@@ -0,0 +1,190 @@
+  indexPattern,
+  setIndexPattern,
+  onError,
+  helpText,
+}: DataViewEditorProps) {
+  useDebounce(
+    () => {
+      dataViewEditorService.setIndexPattern(indexPattern);
+    },
+    250,
+    [indexPattern]
+  );
+  const matchedIndices = useObservable(
+    dataViewEditorService.matchedIndices$,
+    matchedIndicesDefault
+  );
+
+  const matchedReferenceIndices =
+    indexPattern === '' || (indexPattern !== '' && matchedIndices.exactMatchedIndices.length === 0)
+      ? matchedIndices.allIndices
+      : matchedIndices.exactMatchedIndices;
+  const [appendedWildcard, setAppendedWildcard] = useState<boolean>(false);
+
+  const [pageState, updatePageState] = useState({
+    pageIndex: 0,
+    pageSize: 10,
+    sortField: 'name',
+    sortDirection: 'asc',
+  });
+
+  const { onTableChange, pagination } = useTableSettings<MatchedItem>(
+    matchedReferenceIndices.length,
+    pageState,
+    // @ts-expect-error callback will have all the 4 necessary params
+    updatePageState
+  );
+
+  const pageOfItems = useMemo(() => {
+    return matchedReferenceIndices.slice(
+      pagination.pageIndex * pagination.pageSize,
+      (pagination.pageIndex + 1) * pagination.pageSize
+    );
+  }, [pagination.pageSize, pagination.pageIndex, matchedReferenceIndices]);
+
+  const columns = [
+    {
+      field: 'name',
+      name: i18n.translate('xpack.ml.dataDrift.indexPatternsEditor.tableColShard', {
+        defaultMessage: 'Matched indices',
+      }),
+      sortable: false,
+      truncateText: false,
+    },
+  ];
+  const errorMessage = useMemo(() => {
+    if (indexPattern === '')
+      return i18n.translate('xpack.ml.dataDrift.indexPatternsEditor.error.noEmptyIndexPattern', {
+        defaultMessage: 'Index pattern must not be empty.',
+      });
+    if (indexPattern !== '' && matchedIndices.exactMatchedIndices.length === 0) {
+      return mustMatchError;
+    }
+    return undefined;
+  }, [indexPattern, matchedIndices.exactMatchedIndices.length]);
+
+  useEffect(() => {
+    if (onError) {
+      onError(errorMessage);
+    }
+  }, [onError, errorMessage]);
+  const { euiTheme } = useEuiTheme();
+
+  return (
+    <EuiFlexGrid columns={2} gutterSize="none">
+      <EuiFlexItem
+        css={{
+          paddingLeft: euiTheme.size.base,
+          paddingRight: euiTheme.size.base,
+          borderRight: euiTheme.border.thin,
+        }}
+      >
+        <EuiFormRow
+          label={label}
+          error={errorMessage}
+          isInvalid={errorMessage !== undefined}
+          fullWidth
+          helpText={helpText}
+        >
+          <EuiFieldText
+            value={indexPattern}
+            onChange={(e: ChangeEvent<HTMLInputElement>) => {
+              let query = e.target.value;
+              if (query.length === 1 && !appendedWildcard && canAppendWildcard(query)) {
+                query += '*';
+                setAppendedWildcard(true);
+                setTimeout(() => e.target.setSelectionRange(1, 1));
```

Comment:
❓  why do we need a timeout?

Reply:
As discussed in Slack, this ensures the * is appended correctly. I'll leave this here.